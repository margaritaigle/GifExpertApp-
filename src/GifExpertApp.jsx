import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);


  const onAddCategory = ( newCategory ) => {
    // setCategories( cat => [ ...cat, "Valorant" ] );
    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      {/* {titulo} */}
      <h1>GifExpertApp</h1>

      {/* { Input } */}
      <AddCategory
        // addNewCategory={setCategories}
        onNewCategory={ (value) => onAddCategory(value) }
      />

      {/* {Listado de gif} */}
      <ol>
        {categories.map((category) => {
          return <li key={ category } >{ category }</li>
        })}

      </ol>
      {/* {Gif Item} */}
    </>
  );
}

