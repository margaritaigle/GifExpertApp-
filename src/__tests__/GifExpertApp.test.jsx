import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../GifExpertApp";

describe('Testing <GifExpertApp/>', () => { 
    test('should match with snapshot', () => { 
        const { container } = render(<GifExpertApp/>);
        expect(container).toMatchSnapshot();
     });

     test('The category that was already searched, will be rendered only once', async () => {
        render(<GifExpertApp/>)
        const form = screen.getByRole('form');
        const input = screen.getByRole('textbox');
        fireEvent.input( input, { target: {value: 'Gala'}});
        fireEvent.submit( form );
        
        await waitFor( () => {
            fireEvent.input( input, { target: {value: 'Gala'}});
            fireEvent.submit( form );
            const category = screen.getAllByRole('heading', {level: 3, name: 'Gala'});
            expect(category).toHaveLength(1);
        });
        
     });
 });