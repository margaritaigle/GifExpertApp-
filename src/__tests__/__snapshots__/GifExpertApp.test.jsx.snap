// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp/> should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="formulario"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
